README.md

```diff
@@ -14,30 +16,188 @@ C++ Host SDK for Extism
+To add the cpp-sdk to a CMake project:
 
-```bash
-make test
+```cmake
+find_package(extism-cpp)
+target_link_libraries(getting-started extism-cpp)
+```
+
+### Loading a Plug-in
+
+The primary concept in Extism is the [plug-in](https://extism.org/docs/concepts/plug-in). A plug-in is a code module stored in a `.wasm` file.
+
+Plug-in code can come from a file on disk, object storage or any number of places. Since you may not have one handy, let's load a demo plug-in from the web. Let's
+start by creating a main function that loads a plug-in:
+
+```cpp
+#include <extism.hpp>
+
+int main(void) {
+  const auto manifest =
+      extism::Manifest::wasmURL("https://github.com/extism/plugins/releases/"
+                                "latest/download/count_vowels.wasm");
+  extism::Plugin plugin(manifest, true);
+}
 ```
 
-### Installation
+### Calling A Plug-in's Exports
+
+This plug-in was written in Rust and it does one thing, it counts vowels in a string. It exposes one "export" function: `count_vowels`. We can call exports using `Plugin::call`.
+Let's add code to call `count_vowels` to our main func:
+
+```c++
+#include <extism.hpp>
+#include <iostream>
+#include <string>
 
-After building, run the following from the build directory:
+int main(void) {
+  // ...
+
+  const std::string hello("Hello, World!");
+  auto out = plugin.call("count_vowels", hello);
+  std::string response(out.string());
+  std::cout << response << std::endl;
+  // => {"count":3,"total":3,"vowels":"aeiouAEIOU"}
+}
+```
+
+Build it:
 
 ```bash
-sudo make install
+cmake -B build && cmake --build build
+```
+
+Running this should print out the JSON vowel count report:
+
+```shell
+./build/getting-started
+{"count":3,"total":3,"vowels":"aeiouAEIOU"}
+```
+
+All exports have the same interface, optional bytes in and optional bytes out. This plug-in happens to take a string and return a JSON encoded string with a report of results.
+
+### Plug-in State
+
+Plug-ins may be stateful or stateless. Plug-ins can maintain state between calls using variables. Our count vowels plug-in remembers the total number of counted vowels in the "total" key in the result. You can see this by making subsequent calls to the export:
+
+```cpp
+  const std::string hello("Hello, World!");
+  auto out = plugin.call("count_vowels", hello);
+  std::string response(out.string());
+  std::cout << response << std::endl;
+  // => {"count":3,"total":3,"vowels":"aeiouAEIOU"}
+
+  out = plugin.call("count_vowels", hello);
+  response = out.string();
+  std::cout << response << std::endl;
+  // => {"count":3,"total":6,"vowels":"aeiouAEIOU"}
+```
+
+The state variables will persist until the plug-in is freed or reinitialized.
+
+### Configuration
+
+Plug-ins may optionally take a configuration object. This is a static way to configure the plug-in. Our count-vowels plugin takes an optional configuration to change out which characters are considered vowels. Example:
+
+```cpp
+// #include ...
```

Comment:
Might be worth just using `#include <extism.h>` here?

Reply:
Changed to `#include <extism.hpp>`